Fixes #298 - Configuration of app's logger

AntiD2ta opened the pull request:
Fixes #298 

**Changes:**
- Add pipeline for change the level of logs from Settings view
- Add pipeline for change the output of logs from Settings view

**Does this change need to mentioned in CHANGELOG.md?**
yes | no

**Requires testing**
yes

**Comments about testing , should you have some**

AntiD2ta committed:
[ui] refs #298 - Add Radio Buttons to select log output destination and a Combo box to select log level

AntiD2ta committed:
[config] refs #298 - Add default configuration for log level

AntiD2ta committed:
[ui] refs #298 - Add config option for log level

AntiD2ta committed:
[main] [config] refs #298 - Add log level settings

AntiD2ta committed:
[ui] refs #298 - Add TextField for log Level

I would like a ComboBox with fixed options, but I couldn't do it right

AntiD2ta committed:
Merge branch 'develop' into stdevAntiD2ta_t298_Log_events_to_file

AntiD2ta committed:
[util] refs #298 - Keep references of packages loggers

This allows to changes logger properties and propagate this changes across all modules's logger

mauricio1802 committed:
[settings] Add log level as an independent option

AntiD2ta committed:
[ui] refs #298 - Change path to log level configuration

AntiD2ta committed:
[util] refs #298 - Add function to set log's output to a file

AntiD2ta committed:
[util] refs #298 - Add some changes to logging package

Changes:
- Add SetOutput function to set the desired logger's output given a option
- Return error in SetOutputToFile function

AntiD2ta committed:
[config] refs #298 - Add configuration option for log's output

Path: skycoin/log/output

AntiD2ta committed:
[main] refs #298 - Set log's output correctly given option in settings

AntiD2ta committed:
[ui] refs #298 - Add ui to set log's output in Settings

This need to be upgraded

AntiD2ta committed:
[ui] - Remove unnecessary qml log

LALALA log

AntiD2ta committed:
[logging] refs #298 - Create Writer for none option

Writer that write nothing

AntiD2ta committed:
[logging] refs #298 - Add stderr and none options to SetOutput

AntiD2ta committed:
[config] refs #298 - Change default value for log's output to none

AntiD2ta committed:
[models] refs #298 - Apply changed settings related to logs when they are saved

AntiD2ta committed:
[gitignore] - Ignore .log extensions

AntiD2ta committed:
[logging] refs #298 - Refactorize code

Remove SetOutput function and add GetOutputWriter function to return the io.Writer that needs to be given to SetOutputTo function

AntiD2ta committed:
[skycoin] [models] refs #298 - Change output of SkycoinProject's logger 

This occurs when the app's logger change its output

AntiD2ta committed:
[main] - Handle unmarshaling node settings error

AntiD2ta committed:
[makefile] - Improve test-core rule to have a coverage profile

AntiD2ta committed:
[logging] refs #298 - Change permissions of file for logs to 0600

olemis commented on src/models/config.go:
```diff
@@ -129,24 +130,42 @@ func (cs *ConfigSection) getOption(name string, path []string) *KeyValueStorage
 	optV := cs.getOption(opt, path)
 	if optV == nil {
-		//log error
+		//TODO: log error
```
log error

olemis commented on src/models/config.go:
```diff
@@ -129,24 +130,42 @@ func (cs *ConfigSection) getOption(name string, path []string) *KeyValueStorage
 	data, err := json.Marshal(optV.keyValues)
 	if err != nil {
-		//log error
+		//TODO: log error
```
log error

olemis commented on src/models/config.go:
```diff
@@ -129,24 +130,42 @@ func (cs *ConfigSection) getOption(name string, path []string) *KeyValueStorage
+		if name == "level" {
+			level, err := logging.LevelFromString(value)
+			if err != nil {
+				//TODO: log error
```
log error

olemis commented on src/models/config.go:
```diff
@@ -129,24 +130,42 @@ func (cs *ConfigSection) getOption(name string, path []string) *KeyValueStorage
+		} else if name == "output" {
+			writer, err := logging.GetOutputWriter(value)
+			if err != nil {
+				//TODO: log error
```
log error

olemis commented on src/models/config.go:
```diff
@@ -129,24 +130,42 @@ func (cs *ConfigSection) getOption(name string, path []string) *KeyValueStorage
 	optV := cs.getOption(opt, path)
 	if optV == nil {
-		//log error
+		//TODO: log error
```
log error

olemis commented on src/ui/Settings.qml:
```diff
@@ -288,6 +305,65 @@ Page {
+                }
+            }
+        }
+        // RowLayout {
```
remove if not needed

AntiD2ta replied:
That code is for the initial Combo Box that I made, I left it commented for @stdevCrow to fix it

AntiD2ta replied:
Code removed

olemis reviewed (changes requested):
@stdevAntiD2ta more tests need be written

olemis commented:
> @stdevAntiD2ta more tests need be written

for coverage to be reported ok

olemis committed:
Merge remote-tracking branch 'simelo_alt/stdevRulo_t279_improve_coverage' into stdevAntiD2ta_t298_Log_events_to_file

AntiD2ta committed:
[config] - Add logs to package config

AntiD2ta committed:
[models] - Add logs to config view model

AntiD2ta committed:
[logging] - Fix bug

Bug:
Return Stderr writer instead of Stdout when option is stdout

AntiD2ta committed:
[logging] refs #298 - Add initial tests for logging package

AntiD2ta committed:
Merge remote-tracking branch 'simelo/stdevAntiD2ta_t298_Log_events_to_file' into stdevAntiD2ta_t298_Log_events_to_file

AntiD2ta committed:
Merge branch 'stdevRulo_t279_improve_coverage' into stdevAntiD2ta_t298_Log_events_to_file

AntiD2ta committed:
[test] - Solve ineffectual assignments to `val`

AntiD2ta committed:
[test] - Solve assignment to val variable problem

AntiD2ta committed:
[test] - Amend incorrect path to a directory

AntiD2ta committed:
[test] - Solve ineffectual assignment to `val`

olemis committed:
Merge remote-tracking branch 'fc/develop' into stdevAntiD2ta_t298_Log_events_to_file

AntiD2ta committed:
[view] refs #298 - Remove initial implementation of log level Combo Box

AntiD2ta committed:
[test] refs #298 - Fix linter errors  

... reported in https://travis-ci.org/fibercrypto/fibercryptowallet/jobs/640725444#L454
